core/src/main/scala/kafka/common/InterBrokerSendThread.scala

```diff
@@ -84,9 +74,130 @@ abstract class InterBrokerSendThread(name: String,
     }
   }
 
-  def wakeup(): Unit = networkClient.wakeup()
+  private def sendRequests(now: Long): Unit = {
+    var pollTimeout = Long.MaxValue
+    if (unsentRequests.hasRequests) {
+      for (node <- unsentRequests.nodes.asScala) {
+        val requestIterator = unsentRequests.requestIterator(node)
+        while (requestIterator.hasNext) {
+          val request = requestIterator.next
+          if (networkClient.ready(node, now)) {
+            networkClient.send(request, now)
+            requestIterator.remove()
+          } else
+            pollTimeout = if (unsentRequests.hasRequests) AdminClient.DefaultReconnectBackoffMs else Long.MaxValue
```

Comment:
If we compute the `pollTimeout` this way, I think we only need to do it once right before the call to poll instead of on every iteration of this loop. Using `connectionDelay` is actually a little more accurate, so I'd be ok reverting to the previous logic as well.